core/src/services/fs/lister.rs

```diff
@@ -69,7 +72,8 @@ impl oio::List for FsLister<tokio::fs::ReadDir> {
             oio::Entry::new(&rel_path, Metadata::new(EntryMode::Unknown))
         };
 
-        Ok(Some(d))
```

Comment:
Hi, how about using `metakey` to determine whether to call `de.metadata()`, which involves a syscall, instead of just deciding whether to fill in `Metadata`.

If `metakey` is not `default()`, we can call `de.metadata()` to populate the metadata in the entry. We can do this early, before building the `Entry`.

---

Maybe we can even save a `de.file_type()` call.